pkg/cloud/azure/actuators/machineset/controller.go

```diff
@@ -132,30 +125,99 @@ func isInvalidConfigurationError(err error) bool {
 }
 
 func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, error) {
-	providerConfig, err := getproviderConfig(machineSet)
+	klog.Infof("%v: Reconciling MachineSet", machineSet.Name)
+	sku, err := getSKU(r, machineSet)
 	if err != nil {
-		return ctrl.Result{}, mapierrors.InvalidMachineConfiguration("failed to get providerConfig: %v", err)
-	}
-	instanceType, ok := InstanceTypes[strings.ToLower(providerConfig.VMSize)]
-	if !ok {
-		klog.Errorf("Unable to set scale from zero annotations: unknown instance type: %s", providerConfig.VMSize)
-		klog.Errorf("Autoscaling from zero will not work. To fix this, manually populate machine annotations for your instance type: %v", []string{cpuKey, memoryKey, gpuKey})
+		if strings.Contains(err.Error(), "not found in location") {
```

Comment:
I wonder if it's possible to return a typed error instead of doing a string match here 🤔